internal/migrate/versions.go

```diff
@@ -90,6 +97,31 @@ func (migration *Migration) ValidateSteps() error {
 	return nil
 }
 
+// ValidateVersions checks that the version of the migration matches the state of the database
+func (migration *Migration) ValidateVersions(log *zap.Logger) error {
+	for _, step := range migration.Steps {
+		dbVersion, err := migration.getLatestVersion(log, step.DB)
+		if err != nil {
+			return ErrValidateVersionQuery.Wrap(err)
+		}
+
+		if step.Version <= dbVersion {
+			continue
+		} else {
+			return ErrValidateVersionMismatch.New("expected %d <= %d", step.Version, dbVersion)
+		}
+	}
+
+	if len(migration.Steps) > 0 {
```

Comment:
Not sure whether we need logging for this, but I think it's fine to have it as well.

Reply:
i thought it might be useful for testing

Reply:
i changed to `log.Debug`